msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/ADAP3DFeatureDetectionMethod.java

```diff
@@ -149,4 +161,22 @@
     return featureList;
 
   }
+
+  /** {@inheritDoc} */
+  @Override
+  public Float getFinishedPercentage() {
+    return (float) (finalFeatureList.size() == 0?0:1);
```

Comment:
Returning 0 or 1 is not exactly a good way to estimate the progress of the algorithm. Can you come up with a better way?